### PR TITLE
feat(charts): modify custom api filter to include more fields

### PR DESCRIPTION
**Pull request by nytai**
### SUMMARY

similar to https://github.com/apache/incubator-superset/pull/11031 this PR expends the 'search' fields to include slice_name, description, viz_type, or datasource_name

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A
### TEST PLAN

CI, unit tests
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/charts/filters.py [hunk 1]
```diff
@@ -21,15 +21,14 @@
...
 
 
-class ChartNameOrDescriptionFilter(
-    BaseFilter
-):  # pylint: disable=too-few-public-methods
-    name = _("Name or Description")
-    arg_name = "name_or_description"
+class ChartAllTextFilter(BaseFilter):  # pylint: disable=too-few-public-methods
+    name = _("All Text")
+    arg_name = "all_text_charts"
```
Comment (nytai) [inline superset/charts/filters.py:31]: Looks like arg_names share a global namespace. We should add the the resource name to other filters so there's no confusion/collision.

File: superset-frontend/src/views/CRUD/chart/ChartList.tsx [hunk 2]
```diff
@@ -364,7 +364,7 @@ function ChartList(props: ChartListProps) {
       Header: t('Search'),
       id: 'slice_name',
       input: 'search',
-      operator: 'name_or_description',
+      operator: 'all_text_chart',
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/chart/ChartList.tsx:367]: Looks like arg_names share a global namespace? If so, we should add the the resource name to other filters so there's no confusion/collision.

File: superset-frontend/src/views/CRUD/chart/ChartList.tsx [hunk 3]
```diff
@@ -364,7 +364,7 @@ function ChartList(props: ChartListProps) {
       Header: t('Search'),
       id: 'slice_name',
       input: 'search',
-      operator: 'name_or_description',
+      operator: 'chart_all_text',
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/chart/ChartList.tsx:367]: Looks like arg_names share a global namespace? I got a test failure on the previously added `all_text` filter for saved queries. If so, we should add the the resource name to other custom filters so there's no confusion/collision.

Comment (dpgaspar) [reply superset-frontend/src/views/CRUD/chart/ChartList.tsx:367]: Yes, good point!

**Review by dpgaspar — APPROVED** [review]

**Pull request merged**